filters/united-states/new-york/new-york-city/public-art-json-to-json.js

```diff
@@ -1,45 +1,99 @@
-const filter = function (data, stringify) {
+const filter = function (data, std_lib, schema, validator, stringify) {
+    // check for standard library and pull out required functions
+    if (!std_lib) {
+        throw "standard library not provided";
+    }
+    let add_required = std_lib.get("add_required");
+    let add_if_not_null = std_lib.get("add_if_not_null")
+    let remove_if_null = std_lib.get("remove_if_null");
+    let remove_if_empty = std_lib.get("remove_if_empty");
+    let validate_params = std_lib.get("validate_params");
+    let create_dates_template = std_lib.get("create_dates_template");
+    let remove_null_date_fields = std_lib.get("remove_null_date_fields");
+  
+    // validate parameters object
+    schema = validate_params(schema, validator);
+  
+    // convert JSON data to object form
     if (typeof data === 'string' || data instanceof String) {
         data = JSON.parse(data);
     }
-
+  
+    // define new data and errors array
     let new_data = [];
-
+    let errors = [];
+  
+    // iterate through each data entry
     data.map(d => {
         let item = {};
-        item.name = d.title;
-        if (item.name === undefined) {
-            console.log(`Data name not found for art with url ${d.url}`);
+        let skip = false;
+  
+        // add name (required)
+        if (!add_required(item, "name", d, d.title, errors)) {
+            skip = true;
+        }
+  
+        // add coordinates (required)
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, parseFloat(d.longitude), errors)) {
+            skip = true;
         }
-        let coordinates = { longitude: parseFloat(d.longitude), latitude:parseFloat(d.latitude) };
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with url ${d.url}`);
+        if (!add_required(coordinates, "latitude", d, parseFloat(d.latitude), errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
-
-	item.details = {
-		date_created : d.date_created,
-		artwork_type1 : d.artwork_type1,
-		artwork_type2 : d.artwork_type2,
-		material : d.material,
-		location_name : d.location_name,
-		address : d.address,
-		city : d.city,
-		state : d.state,
-		zip_code : d.zip_code,
-		borough : d.borough,
-		subject_keyword : d.subject_keyword,
-		inscription : d.inscription,
-		acquisition : d.acquisition
-	}
-        new_data.push(item);
+        
+        let full_name = "";
+        if (d.primary_artist_last && d.primary_artist_last != "NULL"){
```

Comment:
oh i think here you want to check that the last and first name are both non-null and not equal to "NULL". therefore, you would have four && in this first if condition